### PR TITLE
Integrate Flow to CoroutineTemplate

### PR DESCRIPTION
**Pull request by chornerman**
https://github.com/nimblehq/android-templates/issues/87

## What happened 👀
Integrate `Flow` to CoroutineTemplate
 
## Insight 📝
- Get users data from API to demonstrate `StateFlow`
- Add loading state with ProgressBar to demonstrate `StateFlow`
- Show error to demonstrate `SharedFlow`
- Navigate to Second screen with `SharedFlow`
- Remove `LiveData` related component
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/13492460/122645215-876cca80-d143-11eb-9237-c16b3b6a38bf.mp4

Log according to the above demonstration video
```
2021-07-02 15:48:50.641 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : []
2021-07-02 15:48:51.020 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : [UserEntity(id=1, name=Leanne Graham, username=Bret, email=Sincere@april.biz, address=Address(street=Kulas Light, suite=Apt. 556, city=Gwenborough, zipCode=92998-3874, geo=Geo(latitude=-37.3159, longitude=81.1496)), phone=1-770-736-8031 x56442, website=hildegard.org), UserEntity(id=2, name=Ervin Howell, username=Antonette, email=Shanna@melissa.tv, address=Address(street=Victor Plains, suite=Suite 879, city=Wisokyburgh, zipCode=90566-7771, geo=Geo(latitude=-43.9509, longitude=-34.4618)), phone=010-692-6593 x09125, website=anastasia.net), UserEntity(id=3, name=Clementine Bauch, username=Samantha, email=Nathan@yesenia.net, address=Address(street=Douglas Extension, suite=Suite 847, city=McKenziehaven, zipCode=59590-4157, geo=Geo(latitude=-68.6102, longitude=-47.0653)), phone=1-463-123-4447, website=ramiro.info), UserEntity(id=4, name=Patricia Lebsack, username=Karianne, email=Julianne.OConner@kory.org, address=Address(street=Hoeger Mall, suite=Apt. 692, city=South Elvis, zipCode=53919-4257, geo=Geo(latitude=29.4572, longitude=-164.2990)), phone=493-170-9623 x156, website=kale.biz), UserEntity(id=5, name=Chelsey Dietrich, username=Kamren, email=Lucio_Hettinger@annie.ca, address=Address(street=Skiles Walks, suite=Suite 351, city=Roscoeview, zipCode=33263, geo=Geo(latitude=-31.8129, longitude=62.5342)), phone=(254)954-1289, website=demarco.info), UserEntity(id=6, name=Mrs. Dennis Schulist, username=Leopoldo_Corkery, email=Karley_Dach@jasper.info, address=Address(street=Norberto Crossing, suite=Apt. 950, city=South Christy, zipCode=23505-1337, geo=Geo(latitude=-71.4197, longitude=71.7478)), phone=1-477-935-8478 x6430, website=ola.org), UserEntity(id=7, name=Kurtis Weissnat, username=Elwyn.Skiles, email=Telly.Hoeger@billy.biz, address=Address(street=Rex Trail, suite=Suite 280, city=Howemouth, zipCode=58804-1099, geo=Geo(latitude=24.8918, longitude=21.8984)), phone=210.067.6132, website=elvis.io), UserEntity(id=8, name=Nicholas Runolfsdottir V, username=Maxime_Nienow, email=Sherwood@rosamond.me, address=Address(street=Ellsworth Summit, suite=Suite 729, city=Aliyaview, zipCode=45169, geo=Geo(latitude=-14.3990, longitude=-120.7677)), phone=586.493.6943 x140, website=jacynthe.com), UserEntity(id=9, name=Glenna Reichert, username=Delphine, email=Chaim_McDermott@dana.io, address=Address(street=Dayna Park, suite=Suite 449, city=Bartholomebury, zipCode=76495-3109, geo=Geo(latitude=24.6463, longitude=-168.8889)), phone=(775)976-6794 x41206, website=conrad.com), UserEntity(id=10, name=Clementina DuBuque, username=Moriah.Stanton, email=Rey.Padberg@karina.biz, address=Address(street=Kattie Turnpike, suite=Suite 198, city=Lebsackbury, zipCode=31428-2261, geo=Geo(latitude=-38.2386, longitude=57.2232)), phone=024-648-3804, website=ambrose.net)]
2021-07-02 15:48:53.218 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : [UserEntity(id=1, name=Leanne Graham, username=Bret, email=Sincere@april.biz, address=Address(street=Kulas Light, suite=Apt. 556, city=Gwenborough, zipCode=92998-3874, geo=Geo(latitude=-37.3159, longitude=81.1496)), phone=1-770-736-8031 x56442, website=hildegard.org), UserEntity(id=2, name=Ervin Howell, username=Antonette, email=Shanna@melissa.tv, address=Address(street=Victor Plains, suite=Suite 879, city=Wisokyburgh, zipCode=90566-7771, geo=Geo(latitude=-43.9509, longitude=-34.4618)), phone=010-692-6593 x09125, website=anastasia.net), UserEntity(id=3, name=Clementine Bauch, username=Samantha, email=Nathan@yesenia.net, address=Address(street=Douglas Extension, suite=Suite 847, city=McKenziehaven, zipCode=59590-4157, geo=Geo(latitude=-68.6102, longitude=-47.0653)), phone=1-463-123-4447, website=ramiro.info), UserEntity(id=4, name=Patricia Lebsack, username=Karianne, email=Julianne.OConner@kory.org, address=Address(street=Hoeger Mall, suite=Apt. 692, city=South Elvis, zipCode=53919-4257, geo=Geo(latitude=29.4572, longitude=-164.2990)), phone=493-170-9623 x156, website=kale.biz), UserEntity(id=5, name=Chelsey Dietrich, username=Kamren, email=Lucio_Hettinger@annie.ca, address=Address(street=Skiles Walks, suite=Suite 351, city=Roscoeview, zipCode=33263, geo=Geo(latitude=-31.8129, longitude=62.5342)), phone=(254)954-1289, website=demarco.info), UserEntity(id=6, name=Mrs. Dennis Schulist, username=Leopoldo_Corkery, email=Karley_Dach@jasper.info, address=Address(street=Norberto Crossing, suite=Apt. 950, city=South Christy, zipCode=23505-1337, geo=Geo(latitude=-71.4197, longitude=71.7478)), phone=1-477-935-8478 x6430, website=ola.org), UserEntity(id=7, name=Kurtis Weissnat, username=Elwyn.Skiles, email=Telly.Hoeger@billy.biz, address=Address(street=Rex Trail, suite=Suite 280, city=Howemouth, zipCode=58804-1099, geo=Geo(latitude=24.8918, longitude=21.8984)), phone=210.067.6132, website=elvis.io), UserEntity(id=8, name=Nicholas Runolfsdottir V, username=Maxime_Nienow, email=Sherwood@rosamond.me, address=Address(street=Ellsworth Summit, suite=Suite 729, city=Aliyaview, zipCode=45169, geo=Geo(latitude=-14.3990, longitude=-120.7677)), phone=586.493.6943 x140, website=jacynthe.com), UserEntity(id=9, name=Glenna Reichert, username=Delphine, email=Chaim_McDermott@dana.io, address=Address(street=Dayna Park, suite=Suite 449, city=Bartholomebury, zipCode=76495-3109, geo=Geo(latitude=24.6463, longitude=-168.8889)), phone=(775)976-6794 x41206, website=conrad.com), UserEntity(id=10, name=Clementina DuBuque, username=Moriah.Stanton, email=Rey.Padberg@karina.biz, address=Address(street=Kattie Turnpike, suite=Suite 198, city=Lebsackbury, zipCode=31428-2261, geo=Geo(latitude=-38.2386, longitude=57.2232)), phone=024-648-3804, website=ambrose.net)]
2021-07-02 15:48:54.708 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : [UserEntity(id=1, name=Leanne Graham, username=Bret, email=Sincere@april.biz, address=Address(street=Kulas Light, suite=Apt. 556, city=Gwenborough, zipCode=92998-3874, geo=Geo(latitude=-37.3159, longitude=81.1496)), phone=1-770-736-8031 x56442, website=hildegard.org), UserEntity(id=2, name=Ervin Howell, username=Antonette, email=Shanna@melissa.tv, address=Address(street=Victor Plains, suite=Suite 879, city=Wisokyburgh, zipCode=90566-7771, geo=Geo(latitude=-43.9509, longitude=-34.4618)), phone=010-692-6593 x09125, website=anastasia.net), UserEntity(id=3, name=Clementine Bauch, username=Samantha, email=Nathan@yesenia.net, address=Address(street=Douglas Extension, suite=Suite 847, city=McKenziehaven, zipCode=59590-4157, geo=Geo(latitude=-68.6102, longitude=-47.0653)), phone=1-463-123-4447, website=ramiro.info), UserEntity(id=4, name=Patricia Lebsack, username=Karianne, email=Julianne.OConner@kory.org, address=Address(street=Hoeger Mall, suite=Apt. 692, city=South Elvis, zipCode=53919-4257, geo=Geo(latitude=29.4572, longitude=-164.2990)), phone=493-170-9623 x156, website=kale.biz), UserEntity(id=5, name=Chelsey Dietrich, username=Kamren, email=Lucio_Hettinger@annie.ca, address=Address(street=Skiles Walks, suite=Suite 351, city=Roscoeview, zipCode=33263, geo=Geo(latitude=-31.8129, longitude=62.5342)), phone=(254)954-1289, website=demarco.info), UserEntity(id=6, name=Mrs. Dennis Schulist, username=Leopoldo_Corkery, email=Karley_Dach@jasper.info, address=Address(street=Norberto Crossing, suite=Apt. 950, city=South Christy, zipCode=23505-1337, geo=Geo(latitude=-71.4197, longitude=71.7478)), phone=1-477-935-8478 x6430, website=ola.org), UserEntity(id=7, name=Kurtis Weissnat, username=Elwyn.Skiles, email=Telly.Hoeger@billy.biz, address=Address(street=Rex Trail, suite=Suite 280, city=Howemouth, zipCode=58804-1099, geo=Geo(latitude=24.8918, longitude=21.8984)), phone=210.067.6132, website=elvis.io), UserEntity(id=8, name=Nicholas Runolfsdottir V, username=Maxime_Nienow, email=Sherwood@rosamond.me, address=Address(street=Ellsworth Summit, suite=Suite 729, city=Aliyaview, zipCode=45169, geo=Geo(latitude=-14.3990, longitude=-120.7677)), phone=586.493.6943 x140, website=jacynthe.com), UserEntity(id=9, name=Glenna Reichert, username=Delphine, email=Chaim_McDermott@dana.io, address=Address(street=Dayna Park, suite=Suite 449, city=Bartholomebury, zipCode=76495-3109, geo=Geo(latitude=24.6463, longitude=-168.8889)), phone=(775)976-6794 x41206, website=conrad.com), UserEntity(id=10, name=Clementina DuBuque, username=Moriah.Stanton, email=Rey.Padberg@karina.biz, address=Address(street=Kattie Turnpike, suite=Suite 198, city=Lebsackbury, zipCode=31428-2261, geo=Geo(latitude=-38.2386, longitude=57.2232)), phone=024-648-3804, website=ambrose.net)]
2021-07-02 15:48:56.164 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : [UserEntity(id=1, name=Leanne Graham, username=Bret, email=Sincere@april.biz, address=Address(street=Kulas Light, suite=Apt. 556, city=Gwenborough, zipCode=92998-3874, geo=Geo(latitude=-37.3159, longitude=81.1496)), phone=1-770-736-8031 x56442, website=hildegard.org), UserEntity(id=2, name=Ervin Howell, username=Antonette, email=Shanna@melissa.tv, address=Address(street=Victor Plains, suite=Suite 879, city=Wisokyburgh, zipCode=90566-7771, geo=Geo(latitude=-43.9509, longitude=-34.4618)), phone=010-692-6593 x09125, website=anastasia.net), UserEntity(id=3, name=Clementine Bauch, username=Samantha, email=Nathan@yesenia.net, address=Address(street=Douglas Extension, suite=Suite 847, city=McKenziehaven, zipCode=59590-4157, geo=Geo(latitude=-68.6102, longitude=-47.0653)), phone=1-463-123-4447, website=ramiro.info), UserEntity(id=4, name=Patricia Lebsack, username=Karianne, email=Julianne.OConner@kory.org, address=Address(street=Hoeger Mall, suite=Apt. 692, city=South Elvis, zipCode=53919-4257, geo=Geo(latitude=29.4572, longitude=-164.2990)), phone=493-170-9623 x156, website=kale.biz), UserEntity(id=5, name=Chelsey Dietrich, username=Kamren, email=Lucio_Hettinger@annie.ca, address=Address(street=Skiles Walks, suite=Suite 351, city=Roscoeview, zipCode=33263, geo=Geo(latitude=-31.8129, longitude=62.5342)), phone=(254)954-1289, website=demarco.info), UserEntity(id=6, name=Mrs. Dennis Schulist, username=Leopoldo_Corkery, email=Karley_Dach@jasper.info, address=Address(street=Norberto Crossing, suite=Apt. 950, city=South Christy, zipCode=23505-1337, geo=Geo(latitude=-71.4197, longitude=71.7478)), phone=1-477-935-8478 x6430, website=ola.org), UserEntity(id=7, name=Kurtis Weissnat, username=Elwyn.Skiles, email=Telly.Hoeger@billy.biz, address=Address(street=Rex Trail, suite=Suite 280, city=Howemouth, zipCode=58804-1099, geo=Geo(latitude=24.8918, longitude=21.8984)), phone=210.067.6132, website=elvis.io), UserEntity(id=8, name=Nicholas Runolfsdottir V, username=Maxime_Nienow, email=Sherwood@rosamond.me, address=Address(street=Ellsworth Summit, suite=Suite 729, city=Aliyaview, zipCode=45169, geo=Geo(latitude=-14.3990, longitude=-120.7677)), phone=586.493.6943 x140, website=jacynthe.com), UserEntity(id=9, name=Glenna Reichert, username=Delphine, email=Chaim_McDermott@dana.io, address=Address(street=Dayna Park, suite=Suite 449, city=Bartholomebury, zipCode=76495-3109, geo=Geo(latitude=24.6463, longitude=-168.8889)), phone=(775)976-6794 x41206, website=conrad.com), UserEntity(id=10, name=Clementina DuBuque, username=Moriah.Stanton, email=Rey.Padberg@karina.biz, address=Address(street=Kattie Turnpike, suite=Suite 198, city=Lebsackbury, zipCode=31428-2261, geo=Geo(latitude=-38.2386, longitude=57.2232)), phone=024-648-3804, website=ambrose.net)]
2021-07-02 15:48:57.828 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : []
2021-07-02 15:48:57.936 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : [UserEntity(id=1, name=Leanne Graham, username=Bret, email=Sincere@april.biz, address=Address(street=Kulas Light, suite=Apt. 556, city=Gwenborough, zipCode=92998-3874, geo=Geo(latitude=-37.3159, longitude=81.1496)), phone=1-770-736-8031 x56442, website=hildegard.org), UserEntity(id=2, name=Ervin Howell, username=Antonette, email=Shanna@melissa.tv, address=Address(street=Victor Plains, suite=Suite 879, city=Wisokyburgh, zipCode=90566-7771, geo=Geo(latitude=-43.9509, longitude=-34.4618)), phone=010-692-6593 x09125, website=anastasia.net), UserEntity(id=3, name=Clementine Bauch, username=Samantha, email=Nathan@yesenia.net, address=Address(street=Douglas Extension, suite=Suite 847, city=McKenziehaven, zipCode=59590-4157, geo=Geo(latitude=-68.6102, longitude=-47.0653)), phone=1-463-123-4447, website=ramiro.info), UserEntity(id=4, name=Patricia Lebsack, username=Karianne, email=Julianne.OConner@kory.org, address=Address(street=Hoeger Mall, suite=Apt. 692, city=South Elvis, zipCode=53919-4257, geo=Geo(latitude=29.4572, longitude=-164.2990)), phone=493-170-9623 x156, website=kale.biz), UserEntity(id=5, name=Chelsey Dietrich, username=Kamren, email=Lucio_Hettinger@annie.ca, address=Address(street=Skiles Walks, suite=Suite 351, city=Roscoeview, zipCode=33263, geo=Geo(latitude=-31.8129, longitude=62.5342)), phone=(254)954-1289, website=demarco.info), UserEntity(id=6, name=Mrs. Dennis Schulist, username=Leopoldo_Corkery, email=Karley_Dach@jasper.info, address=Address(street=Norberto Crossing, suite=Apt. 950, city=South Christy, zipCode=23505-1337, geo=Geo(latitude=-71.4197, longitude=71.7478)), phone=1-477-935-8478 x6430, website=ola.org), UserEntity(id=7, name=Kurtis Weissnat, username=Elwyn.Skiles, email=Telly.Hoeger@billy.biz, address=Address(street=Rex Trail, suite=Suite 280, city=Howemouth, zipCode=58804-1099, geo=Geo(latitude=24.8918, longitude=21.8984)), phone=210.067.6132, website=elvis.io), UserEntity(id=8, name=Nicholas Runolfsdottir V, username=Maxime_Nienow, email=Sherwood@rosamond.me, address=Address(street=Ellsworth Summit, suite=Suite 729, city=Aliyaview, zipCode=45169, geo=Geo(latitude=-14.3990, longitude=-120.7677)), phone=586.493.6943 x140, website=jacynthe.com), UserEntity(id=9, name=Glenna Reichert, username=Delphine, email=Chaim_McDermott@dana.io, address=Address(street=Dayna Park, suite=Suite 449, city=Bartholomebury, zipCode=76495-3109, geo=Geo(latitude=24.6463, longitude=-168.8889)), phone=(775)976-6794 x41206, website=conrad.com), UserEntity(id=10, name=Clementina DuBuque, username=Moriah.Stanton, email=Rey.Padberg@karina.biz, address=Address(street=Kattie Turnpike, suite=Suite 198, city=Lebsackbury, zipCode=31428-2261, geo=Geo(latitude=-38.2386, longitude=57.2232)), phone=024-648-3804, website=ambrose.net)]
2021-07-02 15:48:59.858 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : []
2021-07-02 15:48:59.958 27613-27613/co.nimblehq.coroutine.staging D/HomeFragment: Result : [UserEntity(id=1, name=Leanne Graham, username=Bret, email=Sincere@april.biz, address=Address(street=Kulas Light, suite=Apt. 556, city=Gwenborough, zipCode=92998-3874, geo=Geo(latitude=-37.3159, longitude=81.1496)), phone=1-770-736-8031 x56442, website=hildegard.org), UserEntity(id=2, name=Ervin Howell, username=Antonette, email=Shanna@melissa.tv, address=Address(street=Victor Plains, suite=Suite 879, city=Wisokyburgh, zipCode=90566-7771, geo=Geo(latitude=-43.9509, longitude=-34.4618)), phone=010-692-6593 x09125, website=anastasia.net), UserEntity(id=3, name=Clementine Bauch, username=Samantha, email=Nathan@yesenia.net, address=Address(street=Douglas Extension, suite=Suite 847, city=McKenziehaven, zipCode=59590-4157, geo=Geo(latitude=-68.6102, longitude=-47.0653)), phone=1-463-123-4447, website=ramiro.info), UserEntity(id=4, name=Patricia Lebsack, username=Karianne, email=Julianne.OConner@kory.org, address=Address(street=Hoeger Mall, suite=Apt. 692, city=South Elvis, zipCode=53919-4257, geo=Geo(latitude=29.4572, longitude=-164.2990)), phone=493-170-9623 x156, website=kale.biz), UserEntity(id=5, name=Chelsey Dietrich, username=Kamren, email=Lucio_Hettinger@annie.ca, address=Address(street=Skiles Walks, suite=Suite 351, city=Roscoeview, zipCode=33263, geo=Geo(latitude=-31.8129, longitude=62.5342)), phone=(254)954-1289, website=demarco.info), UserEntity(id=6, name=Mrs. Dennis Schulist, username=Leopoldo_Corkery, email=Karley_Dach@jasper.info, address=Address(street=Norberto Crossing, suite=Apt. 950, city=South Christy, zipCode=23505-1337, geo=Geo(latitude=-71.4197, longitude=71.7478)), phone=1-477-935-8478 x6430, website=ola.org), UserEntity(id=7, name=Kurtis Weissnat, username=Elwyn.Skiles, email=Telly.Hoeger@billy.biz, address=Address(street=Rex Trail, suite=Suite 280, city=Howemouth, zipCode=58804-1099, geo=Geo(latitude=24.8918, longitude=21.8984)), phone=210.067.6132, website=elvis.io), UserEntity(id=8, name=Nicholas Runolfsdottir V, username=Maxime_Nienow, email=Sherwood@rosamond.me, address=Address(street=Ellsworth Summit, suite=Suite 729, city=Aliyaview, zipCode=45169, geo=Geo(latitude=-14.3990, longitude=-120.7677)), phone=586.493.6943 x140, website=jacynthe.com), UserEntity(id=9, name=Glenna Reichert, username=Delphine, email=Chaim_McDermott@dana.io, address=Address(street=Dayna Park, suite=Suite 449, city=Bartholomebury, zipCode=76495-3109, geo=Geo(latitude=24.6463, longitude=-168.8889)), phone=(775)976-6794 x41206, website=conrad.com), UserEntity(id=10, name=Clementina DuBuque, username=Moriah.Stanton, email=Rey.Padberg@karina.biz, address=Address(street=Kattie Turnpike, suite=Suite 198, city=Lebsackbury, zipCode=31428-2261, geo=Geo(latitude=-38.2386, longitude=57.2232)), phone=024-648-3804, website=ambrose.net)]
```

### CODE REVIEW

**Comment by Tuubz** [comment]
Awesome! Will review soon 😁

File: CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/base/BaseFragment.kt [hunk 1]
```diff
@@ -67,4 +75,14 @@ abstract class BaseFragment<VB : ViewBinding> : Fragment(), BaseFragmentCallback
...
+
+    protected inline infix fun <T> Flow<T>.bindTo(crossinline action: (T) -> Unit) {
+        with(viewLifecycleOwner) {
+            lifecycleScope.launch {
+                repeatOnLifecycle(Lifecycle.State.STARTED) {
+                    collect { action(it) }
+                }
+            }
+        }
+    }
```
Comment (luongvo) [inline CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/base/BaseFragment.kt:87]: :+1:

File: CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/home/HomeViewModel.kt [hunk 2]
```diff
@@ -1,48 +1,52 @@
...
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
     private val getUsersUseCase: GetUsersUseCase
 ) : BaseViewModel(), Output {
 
-    // TODO: Re-factor LiveData to Flow and remove Event
-    private val _users = MutableLiveData<List<UserEntity>>()
-    override val users: LiveData<List<UserEntity>>
+    private val _users = MutableStateFlow<List<UserEntity>>(listOf())
```
Comment (luongvo) [inline CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/home/HomeViewModel.kt:28]: how about emptyList, listOf should be used for at least I item?

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/home/HomeViewModel.kt (hunk 2 above)
Comment (chornerman) [reply CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/home/HomeViewModel.kt]: Updated 🙏🏼

**Review by luongvo — APPROVED** [review]
Kudos @chornerman, lgtm

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

File: CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/lib/Alias.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package co.nimblehq.coroutine.lib
+
+typealias IsLoading = Boolean
```
Comment (lydiasama) [inline CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/lib/Alias.kt:3]: Can we rename this class to `TypeAlias` instead? What do you think? @chornerman

File: CoroutineTemplate/build.gradle [hunk 4]
```diff
@@ -15,13 +15,14 @@ buildscript {
         androidx_core_ktx_version = '1.3.0'
         androidx_navigation_version = '2.3.4'
         androidx_support_version = '1.1.0'
+        androidx_lifecycle_version = '2.4.0-alpha02'
```
Comment (lydiasama) [inline CoroutineTemplate/build.gradle:18]: Can we keep it alphabetical?

**Comment by lydiasama** [comment]
@chornerman Could you please add `[#86]` at your PR title also?
And if you have 2 parts of it, can you add [Part1/2] for helping us on reviewing? 🥺

**Comment by chornerman** [comment]
@lydiasama I believe the #86 is for Jetpack Compose related PR, also this PR has only one part 🙏🏼 , but I will ensure that both Jetpack Compose PRs have #86 in the description and will add `[PartX/2]` to them.

File: CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/lib/Alias.kt (hunk 3 above)
Comment (chornerman) [reply CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/lib/Alias.kt]: Resolved, thank you 🙏🏼

File: CoroutineTemplate/build.gradle (hunk 4 above)
Comment (chornerman) [reply CoroutineTemplate/build.gradle]: Resolved, thank you 🙏🏼

**Review by lydiasama — APPROVED** [review]

**Pull request merged**